packages/zowe-explorer/src/extension.ts

```diff
@@ -771,7 +785,13 @@ function initJobsProvider(context: vscode.ExtensionContext) {
         vscode.commands.registerCommand("zowe.jobs.setPrefix", (job) => jobActions.setPrefix(job, jobsProvider))
     );
     context.subscriptions.push(
-        vscode.commands.registerCommand("zowe.jobs.removeJobsSession", (job) => jobsProvider.deleteSession(job))
+        vscode.commands.registerCommand("zowe.jobs.removeJobsSession", (job, jobList) => {
+            let selectedNodes = getSelectedNodeList(job, jobList);
+            selectedNodes = selectedNodes.filter((element) => contextuals.isJobsSession(element));
```

Comment:
This can be `const`:

```js
const selectedNodes = getSelectedNodeList(job, jobList).filter((element) => contextuals.isJobSession(element));
```

Reply:
Thanks @traeok for reviewing it . Although this variable is immutable I'm reassigning value of at :
`selectedNodes = selectedNodes.filter((element) => contextuals.isJobsSession(element));`
          It can't be const variable because of this.

Reply:
If you join the declaration and assignment, you *should* be able to filter the `getSelectedNodeList` and store the result as const, as long as you don't modify `selectedNodes` later in the function. `getSelectedNodeList` will always return an array, but if its empty the filter will just return early with the empty array - since selectedNodes isn't modified after the filter, we should be able to store the filter result as a const as long as we chain the functions:

```js
const selectedNodes = getSelectedNodeList(job, jobList).filter((element) => contextuals.isJobsSession(element));
for (const node of selectedNodes) {
    // ...
}
```

But as I mentioned earlier, this is just a suggestion and doesn't affect the overall functionality - so no worries about prioritizing these changes.